src/diffusers/schedulers/scheduling_dpmsolver_singlestep.py

```diff
@@ -251,7 +255,14 @@ def set_timesteps(self, num_inference_steps: int, device: Union[str, torch.devic
         self.timesteps = torch.from_numpy(timesteps).to(device)
         self.model_outputs = [None] * self.config.solver_order
         self.sample = None
-        self.orders = self.get_order_list(num_inference_steps)
+
+        if not self.config.lower_order_final and num_inference_steps % self.config.solver_order != 0:
+            logger.warn(
+                "Changing scheduler {self.config} to have `lower_order_final` set to True to handle uneven amount of inference steps. Please make sure to always use an even number of `num_inference steps when using `lower_order_final=True`."
+            )
+            self.register_to_config(lower_order_final=True)
+
+        self.order_list = self.get_order_list(num_inference_steps)
 
     # Copied from diffusers.schedulers.scheduling_ddpm.DDPMScheduler._threshold_sample
```

Comment:
Does this still hold?

Reply:
Yes, I think so! It's a different function I think (`_threshold_sample` vs. `step`) :-)

Reply:
What I mean is, the https://github.com/huggingface/diffusers/blob/7f6373d2640e25ae99d520eef0b94649e8134dfe/src/diffusers/schedulers/scheduling_ddpm.py#L278 doesn't seem to have:

```
order = order if self.model_outputs[-order] is not None else order - 1
```

Is that allowed in `# Copied from ...` statements?

Reply:
```
order = order if self.model_outputs[-order] is not None else order - 1
```

was added to the `def step(...)` function not the `def _threshold_sample` function :-) think the git diff review made it hard to read ;-)

Reply:
Ah! Sorry.